arangod/Aql/GraphNode.cpp

```diff
@@ -202,20 +253,27 @@ GraphNode::GraphNode(ExecutionPlan* plan, ExecutionNodeId id,
         THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_ARANGO_COLLECTION_TYPE_INVALID,
```

Comment:
The error message could be misleading if the collection cannot be found. Then we should rather return TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND and an appropriate message.
But I don't insist on making this change.